src/main/java/org/opensearch/neuralsearch/processor/TextImageEmbeddingProcessor.java

```diff
@@ -183,8 +183,8 @@ Map<String, String> buildMapWithKnnKeyAndOriginalValue(final IngestDocument inge
             if (!sourceAndMetadataMap.containsKey(originalKey)) {
                 continue;
             }
-            if (!(sourceAndMetadataMap.get(originalKey) instanceof String)) {
-                throw new IllegalArgumentException("Unsupported format of the field in the document, value must be a string");
+            if (!(sourceAndMetadataMap.get(originalKey) instanceof String) || ((String) sourceAndMetadataMap.get(originalKey)).isEmpty()) {
```

Comment:
```suggestion
((String) sourceAndMetadataMap.get(originalKey)).isEmpty()) {
```

It is not a safe type cast. In case of null string it will throw an exception.

Reply:
`null instanceof String` returns false even if the reference type fo the null is String (see https://docs.oracle.com/javase/specs/jls/se11/html/jls-15.html#jls-15.20.2). So the boolean short circuit should make this cast safe. But I also suggest `StringUtils.isNotBlank()` NPEs just to be safe.